script.js

```diff
@@ -0,0 +1,487 @@
+  }
+  return `<span class="key">${letter}</span>`;
+}
+
+function addKeysToRow(row, lettersSet) {
+  const rowElem = row;
+  const rowKeysMurkup = lettersSet.map((letter) => getMupkupFromLetter(letter)).join('');
+  rowElem.innerHTML = rowKeysMurkup;
+}
+
+function getSubarrayForRow(lettersSet, rowIndex) {
+  let subarrayStartPosition = 0;
+  if (rowIndex > 0) {
+    subarrayStartPosition = ROWS_SIZE[rowIndex - 1];
+  }
+  const subarrayEndPosition = ROWS_SIZE[rowIndex];
+  return lettersSet.slice(subarrayStartPosition, subarrayEndPosition);
+}
+
+function fillWithLetters() {
+  const rows = templateElem.querySelectorAll('.row');
+  const lettersSet = lettersList[`${lang}Keys`];
+
+  rows.forEach((row, index) => addKeysToRow(row, getSubarrayForRow(lettersSet, index)));
+}
+
+function isRegularLetter(innerText) {
+  return innerText.length === 1;
+}
+
+function textTyping(innerText) {
+  const cursorPosition = textInput.selectionStart;
+  const textBeforeCursor = textInput.value.slice(0, cursorPosition);
+  const textAfterCursor = textInput.value.slice(textInput.selectionEnd);
+  textInput.value = textBeforeCursor + innerText + textAfterCursor;
+  textInput.selectionStart = cursorPosition + 1;
+  textInput.selectionEnd = cursorPosition + 1;
+}
+
+function isLangChangingPressed(pressedKeysArray) {
+  const pressedCodeList = pressedKeysArray.map((key) => key.innerText);
+  return LANG_CHANGING_KEYS[0].every((key) => pressedCodeList.indexOf(key) !== -1)
+  || LANG_CHANGING_KEYS[1].every((key) => pressedCodeList.indexOf(key) !== -1);
+}
+
+function deleteSelectedDiapason() {
+  textInput.value = textInput.value.slice(0, textInput.selectionStart)
+  + textInput.value.slice(textInput.selectionEnd);
+}
+
+function isMultiContentKey(index) {
+  if (index <= MULTI_CONTENT_KEYS) {
+    return true;
+  }
+  return false;
+}
+
+const specialKeysHandlers = {
+  backspace() {
+    if (textInput.value.length < 1) {
+      return;
+    }
+    if (textInput.selectionStart !== textInput.selectionEnd) {
+      const cursorPosition = textInput.selectionStart;
+      deleteSelectedDiapason();
+      textInput.selectionStart = cursorPosition;
+      textInput.selectionEnd = cursorPosition;
+      return;
+    }
+    if (textInput.selectionEnd > 0) {
+      const cursorPosition = textInput.selectionStart;
+      textInput.value = textInput.value.slice(0, cursorPosition - 1)
+      + textInput.value.slice(cursorPosition);
+      textInput.selectionStart = cursorPosition - 1;
+      textInput.selectionEnd = cursorPosition - 1;
+    }
+  },
+  tab() {
+    textTyping('\t');
+  },
+  enter() {
+    textInput.value += '\n';
+  },
+  del() {
+    if (textInput.selectionStart !== textInput.selectionEnd) {
+      const cursorPosition = textInput.selectionStart;
+      deleteSelectedDiapason();
+      textInput.selectionStart = cursorPosition;
+      textInput.selectionEnd = cursorPosition;
+      return;
+    }
+    if (textInput.selectionEnd <= textInput.value.length) {
```

Comment:
What if **textInput.value** will be undefined ? Undefined.length will cause a type error. 

You can fix it like this: 
if (textInput.selectionEnd <= textInput && textInput.value && textInput.value.length) 

or in ES7 there is a new operator ? that do the same check: 
textInput?.value?.length.

You can read about it here: https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Operators/Optional_chaining. 

Use the first example here, please. You can move very long condition to a variable, for example:

let isLessThanSelectionEnd = textInput.selectionEnd <= textInput && textInput.value && textInput.value.length;

And after that use it like this: 

 if(isLessThanSelectionEnd) {...}